src/GraphQL/Utilities/Federation/FederatedSchemaPrinter.cs

```diff
@@ -68,7 +64,10 @@ public override string PrintObject(IObjectGraphType type)
 
             var extended = isExtension ? "extend " : "";
 
-            return FormatDescription(type.Description) + "{1}type {2}{3}{4} {{{0}{5}{0}}}".ToFormat(Environment.NewLine, extended, type.Name, implementedInterfaces, federatedDirectives, PrintFields(type));
+            if (type.Fields.Any(x => !IsFederatedType(x.ResolvedType!.GetNamedType().Name)))
```

Comment:
`FederatedSchemaPrinter` overrides `PrintFields` so this check correlates to `PrintFields` implementation.